src/transformers/models/dpt/modeling_flax_dpt.py

```diff
@@ -0,0 +1,1236 @@
+        output_hidden_states: bool = False,
+        return_dict: bool = True,
+        labels: Optional[jnp.ndarray] = None,
+    ):
+
+        hidden_states = self.embeddings(pixel_values, deterministic=deterministic)
+
+        outputs = self.encoder(
+            hidden_states,
+            deterministic=deterministic,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        hidden_states = outputs[0]
+        hidden_states = self.layernorm(hidden_states)
+        pooled = self.pooler(hidden_states) if self.add_pooling_layer else None
+
+        if not return_dict:
+            # if pooled is None, don't return it
+            if pooled is None:
+                return (hidden_states,) + outputs[1:]
+            return (hidden_states, pooled) + outputs[1:]
+
+        return FlaxBaseModelOutputWithPooling(
+            last_hidden_state=hidden_states,
+            pooler_output=pooled,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+
+@add_start_docstrings(
+    """
+    DPT Model with a semantic segmentation head on top e.g. for ADE20k, CityScapes.
+    """,
+    DPT_START_DOCSTRING,
+)
+class FlaxDPTModel(FlaxDPTPreTrainedModel):
+    module_class = FlaxDPTModule
+
+
+class FlaxDPTConvCollection(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.convs = [
+            nn.Conv(self.config.fusion_hidden_size, kernel_size=(3, 3), padding=1, use_bias=False, name=str(i))
+            for i in range(len(self.config.neck_hidden_sizes))
+        ]
+
+    def __call__(self, features):
+        return [self.convs[i](feature) for i, feature in enumerate(features)]
+
+
+class FlaxDPTNeck(nn.Module):
+    config: DPTConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        # postprocessing
+        self.reassemble_stage = FlaxDPTReassembleStage(self.config, self.dtype)
+        self.convs = FlaxDPTConvCollection(self.config, self.dtype)
+        # fusion
+        self.fusion_stage = FlaxDPTFeatureFusionStage(self.config)
+
+    def __call__(self, hidden_states):
+        if not isinstance(hidden_states, list):
+            raise ValueError("hidden_states should be a list of tensors")
+
+        if len(hidden_states) != len(self.config.neck_hidden_sizes):
+            raise ValueError("The number of hidden states should be equal to the number of neck hidden sizes.")
+
+        # postprocess hidden states
+        features = self.reassemble_stage(hidden_states)
+
+        features = self.convs(features)
+
+        # fusion blocks
+        output = self.fusion_stage(features)
+
+        return output
+
+
+class FlaxDPTUpsample(nn.Module):
+    scale: int = 2
+    method: str = "bilinear"
+
+    def setup(self):
+        pass
+
+    def __call__(self, x, output_size=None):
+        if output_size is None:
```

Comment:
Wonder if we can improve clarity here by aligning with PT?
```suggestion
    def __call__(self, x, align_corners=True):
        if not align_corners:
```